python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -186,14 +208,21 @@ def ptr(self) -> int:
         """Device pointer to the start of the buffer."""
         return self._ptr
 
+    @property
+    def mutable_ptr(self) -> int:
```

Comment:
> it's just that accessing `ptr` and mutating the result may lead to unexpected higher-level behavior e.g. incorrect COW behavior, right?

Correct, for `Buffer` both can be used to mutate. Do you think a utility would be appropriate instead of `mutable_ptr`?(If yes, what should we call it?). For context, here was the discussion https://github.com/rapidsai/cudf/pull/11718#discussion_r1049989571 where we introduced `mutable_ptr`.

cc: @shwina

Reply:
I think it would be worth us meeting (with other interested parties like @wence- and @shwina) to discuss the minimal changes needed to address the issues with confusing APIs introduced in this PR.